ddl/table.go

```diff
@@ -267,8 +268,7 @@ func (w *worker) onRecoverTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver in
 		}
 
 		failpoint.Inject("mockRecoverTableCommitErr", func(val failpoint.Value) {
-			if val.(bool) && mockRecoverTableCommitErrOnce {
-				mockRecoverTableCommitErrOnce = false
+			if val.(bool) && atomic.CompareAndSwapUint32(&mockRecoverTableCommitErrOnce, 0, 1) {
```

Comment:
how about change
https://github.com/pingcap/tidb/blob/27a56180bd20a4151458ea60cd3667f2de598264/ddl/serial_test.go#L443
to `1*return(true)->return(false)` instead of using `mockRecoverTableCommitErrOnce`.

Reply:
CI fail if a change to this way @jackysp @zimulala

Reply:
...